### PR TITLE
Security (Officer and detective) will very rarely be antagonists.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Edit:
Ill change this to changeling since its 100x better than sec traitors.
Security should have at least 1 antagonist role that they can be and I think changeling is the best (and a good one) for it.

## About The Pull Request

Very rarely (1/20 rounds), a member of security (Detective or Security off.) will be a traitor.
They will start with 16 TC, however are in a very powerful position for gaining trust and items.
This will not spawn without a HOS or Warden, see below.
This can only spawn on gamemodes that spawn traitors normally (traitorbros, traitor, traitorchan).

## Why It's Good For The Game

This removes the 100% guaranteed protection from security officers or detectives. While still incredibly rare, the small chance is enough to mean that the HOS and warden will have a more active role in monitoring the behaviour of security and ensuring that gear isn't given out excessively.
While security being full on shitsec is of course against the rules and will be punished by the admins, there are cases which are a lot more borderline that should be dealt IC. Currently as a HOS, many players (myself included) find it difficult to deal with situations like this in an IC manner with the knowledge that you know they aren't bad or simply don't care since the antagonists are now defeated.
This PR aims to increase the responsibility of the HOS' job in managing the actions of security and makes security more in line with other members of the crew in that they will too need to be monitored for suspicious activity.
Additionally warden will have a larger responsibility in determining that a threat is great enough to give out weapons to all members of security, as you could be empowering the very people you aim to defeat.
The main goal is more to make it less abnormal for HOS or wardens to check up on security, check them for illegal items and make sure they aren't doing anything they shouldn't be rather than allowing for security to go on a rampage as the ultimate shitsec.

This shouldn't act as a nerf to security, the role itself is extremely rare and mainly makes it easier for HOS and the warden to inspect their own security members and ensure good behaviour all round.
Regular security officers sneaking contraband from captured traitors will no longer get a free all-clear pass on the premise that they are security so have to be good but instead will need to be questioned like any other member of the crew.

## Changelog
:cl:
refactor: Adds in the infiltrator antagonist role, a regular traitor with 16 TC however will be detective or security officer.
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
There is no way this can end well.

**Comment by Fresh839** [comment]
As much as i love this PR in regards to the security officer change i will have to say that this will just make the detective even more ignored than he already is. Validhunting security officers/HOS (which are not a minority by the way) can see this change as a reason to not trust the detective. As he could be faking his evidence and all that.
This is in my opinion a hard nerf to detective. It is already very hard to establish yourself right now if you want to be a competent/actual detective and this change doesn't help this but rather increases the likelyhood of security completely ignoring you as a detective even more. Just my two cents here. I like the idea in regards to security officers though. Even though it could probably end in a horrible mess and terrible experiences for HOS players.

**Comment by mcmeiler** [comment]
There is no way that giving a traitor free security gear at roundstart could end well.

**Comment by PowerfulBacon** [comment]
> 
> 
> As much as i love this PR in regards to the security officer change i will have to say that this will just make the detective even more ignored than he already is. Validhunting security officers/HOS (which are not a minority by the way) can see this change as a reason to not trust the detective. As he could be faking his evidence and all that.
> This is in my opinion a hard nerf to detective. It is already very hard to establish yourself right now if you want to be a competent/actual detective and this change doesn't help this but rather increases the likelyhood of security completely ignoring you as a detective even more. Just my two cents here. I like the idea in regards to security officers though. Even though it could probably end in a horrible mess and terrible experiences for HOS players.

All the more reason to write conclusive reports summing up evidence and using the print outs from the forensics scanner. Detectives often get ignored because they keep the evidence to themselves and just tell sec to arrest a guy who then gets released when they find no traitor items on them. Writting up reports, taking photos and creating cases against someone to say that this person is conclusively guilty is something detectives should be doing anyway. I don't think detectives will be given too much distrust, the main problem I can think of is who will detect the crimes of the detective.

**Comment by PowerfulBacon** [comment]
> 
> 
> There is no way that giving a traitor free security gear at roundstart could end well.

It's more of giving security traitor status than giving traitors security gear
 - They spawn with less TC
 - They are significantly rarer than normal traitors
 - They require a HOS or warden to spawn
The traitor part is more the minor part considering their low rate, the biggest part is changing the mentality that security are always safe

File: code/modules/antagonists/traitor/traitor_spawner.dm [hunk 1]
```diff
@@ -31,3 +29,32 @@
...
+	protected_jobs = get_all_jobs()
+	protected_jobs -= list("Detective", "Security Officer")
+
+/datum/antagonist/traitor/infiltrator
+	name = "Infiltrator"
+	telecrystals = 16
+
+/datum/antagonist/traitor/infiltrator/on_gain()
+	. = ..()
+	to_chat(owner.current, "<span class='syndradio italics'>It's taken a lot of work from a lot of people to get you in this possition. Don't mess this up, we are counting on you.</span>")
```
Comment (Raven-Industries) [inline code/modules/antagonists/traitor/traitor_spawner.dm:59]: Spelling error here, unless it's a lizardperson giving you this "possition".

**Comment by St0rmC4st3r** [comment]
Now we should add some role-specific items for seccies

**Comment by Mat05usz** [comment]
First of all, if anything they should be given no TCs at all, I would gladly dump all of mine just to get sec status as it makes so much easier. 

Second, forcing it to spawn only when there is a HoS won't really matter, as security is a shit department usually filled with people that have brain capacity comparable to a dense monkey.

Third, currently security is metaprotected by rules so if you allow antags to spawn with this metaprotection it's gonna result in a shitshow with people not knowing how to behave as to not get banned.

While I undertand the reasoning and I kinda like the idea, it will not work. Not with community in its current shape nor with rules.

**Comment by Raven-Industries** [comment]
> Third, currently security is metaprotected by rules so if you allow antags to spawn with this metaprotection it's gonna result in a shitshow with people not knowing how to behave as to not get banned.

What rules are you referring to here?

**Comment by mcmeiler** [comment]
>They spawn with less TC

I don't see how giving a member of sec antag status and an uplink with a whopping 4 less TC is a good idea. 16 still buys 99% of options from the uplink. Either give them less than 6 TC or none at all.

>They require a HOS or warden to spawn

Players will meta this to shit, saying "oh we have no hos/warden so sec cant be antag".

**Comment by Mat05usz** [comment]
> > Third, currently security is metaprotected by rules so if you allow antags to spawn with this metaprotection it's gonna result in a shitshow with people not knowing how to behave as to not get banned.
> 
> What rules are you referring to here?

https://wiki.beestation13.com/view/Escalation_Policy

Security clause. Also try and stun + cuff secoff, wait for him to ahelp and get bwoinked for 'bad rp'.

**Comment by mcmeiler** [comment]
> 
> 
> > > Third, currently security is metaprotected by rules so if you allow antags to spawn with this metaprotection it's gonna result in a shitshow with people not knowing how to behave as to not get banned.
> > 
> > 
> > What rules are you referring to here?
> 
> https://wiki.beestation13.com/view/Escalation_Policy
> 
> Security clause. Also try and stun + cuff secoff, wait for him to ahelp and get bwoinked for 'bad rp'.

This. Unless this PR brings rule changes as well (which it very likely does not), this doesn't even work in the slightest since you will immediately get in trouble for defending yourself against an infiltrator.

**Comment by Mat05usz** [comment]
> > > > Third, currently security is metaprotected by rules so if you allow antags to spawn with this metaprotection it's gonna result in a shitshow with people not knowing how to behave as to not get banned.
> > > 
> > > 
> > > What rules are you referring to here?
> > 
> > 
> > https://wiki.beestation13.com/view/Escalation_Policy
> > Security clause. Also try and stun + cuff secoff, wait for him to ahelp and get bwoinked for 'bad rp'.
> 
> This. Unless this PR brings rule changes as well (which it very likely does not), this doesn't even work in the slightest since you will immediately get in trouble for defending yourself against an infiltrator.

Eh I don't think that you would be banned for defending against the impostor (I had to, sorry), but since sec is already full of shit  you would have a lot of trouble determining who is sec and who isn't. Now you can at least feel 'safe' in a way that even if the reason for them batoning you is bullshit, they will at least probably bring you to brig so you don't have to turn to your primal instincts and robust them back  in case they are a tot (which may result in a ban if he's not one because 'how dare you baton the officer back, you stinky powergamer').


To tldr what I mean:
Sec batongs you: you may only batong back, but if anyone sees you batoning redshit you can be sure as fuck they'll help him out.
Tot batongs you: you may crit him and probably noone will care.
Sec tot batongs you: Basically what would happen if regular sec batonged you but you have no way of knowing if he's a tot or not, people are used to hearing that redshit X is tot when someone is being cuffed, then he strips your headset and finishes you in maint.

And if you make mistake and think that redshit is a tot, when he's not, admin comes and batongs you because cringe rules.

**Comment by PowerfulBacon** [comment]
> Security Clause: Security officers, the head of security and captain may escalate and skip to lethal force in accordance with Space Law's use of deadly force. Non-antagonists may not use lethal force against security until after they have used it against you without being baited or forced into using it. This means critting or killing security that is doing their job properly is always over-escalation for non-antagonists. As a non-antagonist, you are expected to disable/bamboozle security and run away or submit to the arrest.

You can defend yourself from arrests with non lethal force. You can stun officers for trying to arrest you and won't get punished if the officer is attacking you for no reason.

**Comment by Mat05usz** [comment]
> > Security Clause: Security officers, the head of security and captain may escalate and skip to lethal force in accordance with Space Law's use of deadly force. Non-antagonists may not use lethal force against security until after they have used it against you without being baited or forced into using it. This means critting or killing security that is doing their job properly is always over-escalation for non-antagonists. As a non-antagonist, you are expected to disable/bamboozle security and run away or submit to the arrest.
> 
> You can defend yourself from arrests with non lethal force. You can stun officers for trying to arrest you and won't get punished if the officer is attacking you for no reason.

I did explain what I mean further in an edit to a comment above.

**Comment by PowerfulBacon** [comment]
I can make it so they only get steal objectives and have a kill objective similar to IAA / incursion where you know someone is after you and they are a traitor.

**Comment by Mat05usz** [comment]
> I can make it so they only get steal objectives and have a kill objective similar to IAA / incursion where you know someone is after you and they are a traitor.

That would still be giving an unfair advantage to one lucky tot while also basically cucking the target. Just don't buy anything for your TCs and unless you have like 13 IQ, you will find a way to cuck your target and HoS will probably give you a medal for robusting da valid.

And steal objectives are fucking lame anyways cause they give strong incentive to play stealth.

**Comment by Kapu1178** [comment]
Jesus. Fucking. Christ. 

NO.

Bee players, let alone BEE SEC, CANNOT be trusted with antag status. This makes HoS a living hell to play, and will just cause a massive shitstorm with the rules.

**Comment by PowerfulBacon** [comment]
> 
> 
> Jesus. Fucking. Christ.
> 
> NO.
> 
> Bee players, let alone BEE SEC, CANNOT be trusted with antag status. This makes HoS a living hell to play, and will just cause a massive shitstorm with the rules.

One of the many reasons I wanted to make something like this was to make it easier for the HOS to manage and question their security officers. HOS is essentially security officer+ right now and doesn't need to keep track of the actions of their officers.
I would like to see how this plays out in action, since the changes to brig physician were not disasterous and actually did a lot of good.

Unfortunately due to the nature of security there will always be downsides to whatever there is done with them. Poor behaviour is heavilly encouraged in security and having the bad sec officer 1/20 * the probability of traitor gamemodes rounds instead of every other round is a big improvement, since you won't be able to keep contraband for yourself, beat cuffed people to death and other shitsec behaviours without some questioning from the HOS.

I really think that life as HOS will be a lot easier when the mentality isn't 'why are you questioning me, you know I can't be bad'.

You can argue that they get sec gear and are in a position of trust, but how is it any different from almost every other job.
Engineer traitors have the ability to destroy the powernet, destroy the station, why aren't they mindshielded?
Scientists can freely make bombs, summon hostile creatures, create mechs, create research, why aren't they metaprotected.
Medical doctors handle people when they physically cannot fight back at all, have access to surgery where they can freely brainwash and they are fine not being metaprotected.
Security gets access to some stun weapons, cuffs (And usually every single ballistic gun in cargo which will hopefully change after this).
Every job is as critical as others, and a lot of other jobs require most trust than sec (medical doctor etc.) but work fine without metaprotection.

**Comment by Mat05usz** [comment]
Noone questions sec batoning, or sometimes even killing, X in a place Y, that's what makes it different from other jobs.

Really, just think of any interaction/brawl/fight between two people and how would everyone around react when one of them is sec vs when both are regular crew. If I saw sec batoning someone I'd probably just mind my own business and not care, while if I saw two people fighting I'd probably call sec and point out exactly where that is. And I'm someone who follows the rules, a lot of people would probably even help the seccie, that's how powerful tot as sec can be.

> Engineer traitors have the ability to destroy the powernet, destroy the station, why aren't they mindshielded?

Because power net is easily repairable since there is usually a shittone of engies. Also due to our cringe rules murderbone is not allowed so noone even thinks about engies doing destruction.

> Scientists can freely make bombs, summon hostile creatures, create mechs, create research, why aren't they metaprotected.

Can't bone, can't bone, can't bone (also can't powergame) and research is for the most part harmless.

> Medical doctors handle people when they physically cannot fight back at all, have access to surgery where they can freely brainwash and they are fine not being metaprotected.

Yeah but there is usually fucktone of people in medbay, that's true that doctors have it easier when it comes to brainwashing but 🤷 

> Security gets access to some stun weapons, cuffs

Why are you trying to make the fact that sec has access to most deadly weapons there are non significant? You know damn well how robust even roundstart sec gear is.


Also as I said before, security is not only metaprotected by the fact that they can't  be antags, they are even more protected by rules, and by extension admins.


I understand the intention behind this, and I do believe that some change is needed, but this is absolutely no way to go and will cause more harm than good. Also changes to brig phys were not disastrous because that's a single person, yet many people have died because of thinking that brig phys is still metaprotected, and probably still are.

**Comment by KitsuneQueen** [comment]
Change is fine. As a sec main, I go solo and don't trust anyone anyway. Hopefully this change will make more players play correctly :)

**Comment by PowerfulBacon** [comment]
> Noone questions sec batoning, or sometimes even killing, X in a place Y, that's what makes it different from other jobs.

Yes, its different from other jobs at the moment because there is 0 chance security officers can be antagonists, they are 100% safe.

> Really, just think of any interaction/brawl/fight between two people and how would everyone around react when one of them is sec vs when both are regular crew. If I saw sec batoning someone I'd probably just mind my own business and not care, while if I saw two people fighting I'd probably call sec and point out exactly where that is. And I'm someone who follows the rules, a lot of people would probably even help the seccie, that's how powerful tot as sec can be.

Just because you can put someone in cuffs doesn't necessarilly make it powerful. You can easilly arrest someone in front of other people, but then you have a lot of eyes on you making it extremely difficult to just out right kill them there and then. The only way you can really get rid of people without a ton of people seeing is arresting them in maints or alone, where nobody will be there to assist you.
And what's more powerful than fighting your target when they are unconcious and can't fight back, surely doctors should be protected if its about the role being too easy.

> Because power net is easily repairable since there is usually a shittone of engies. Also due to our cringe rules murderbone is not allowed so noone even thinks about engies doing destruction.

There are usually a shit ton of seccies too, someone being dead is easilly fixable and as a sec its basically impossible to have your sensors off without the warden noticing.
Due to the murderboning roles its not like they will be going on a rampage either.

> Scientists can freely make bombs, summon hostile creatures, create mechs, create research, why aren't they metaprotected.
> Can't bone, can't bone, can't bone (also can't powergame) and research is for the most part harmless.

You can still use bombs to assassinate targets and summon hostile creatures around them. Hostile creatures would probably not be considered murderboning if put next to your target and as long as the bomb isn't doing insane collateral damage and in a critical area it probably wouldn't be considered murderboning either.
Mechs are weapons not tools of mass destruction and antagonists can powergame.

> Yeah but there is usually fucktone of people in medbay, that's true that doctors have it easier when it comes to brainwashing but 🤷

Theres usually a fuck ton of people in security too.

**Comment by PowerfulBacon** [comment]
Could make them changelings instead of traitors.
Would make more sense for a changeling to infiltrate security and they have better abilities for this.

**Comment by Crossedfall** [comment]
Maybe I'm reading into this wrong or maybe I have a bad perception on the change as a whole... But, it really kind of comes off as, "I don't know how to roleplay certain situations, so I codified a solution to make it easier."

Which.. Maybe that's fine? But I don't know if that's enough to really sell me on the idea

**Comment by PowerfulBacon** [comment]
I just had a far better idea

**Comment by EvilDragonfiend** [comment]
Being a sec is already a strong feature for taking 4 TC only. I think I can agree with some conditions.

- 1/40 chance to turn on a sec antag mode.
- as long as sec antag is turned on, secs are allowed to be spawned as a traitor.
- After 30 minutes later, CentCom sends an official announcement to the bridge.
   - "We seriously show our apology to all of our finest NanoTrasen heads about a critical vulnerable issue of our loyalty training course recently. Due to this, It appears that a few improper crews are assigned as a security officer or a detective of your station. We're doing our best to fix the vulnerable issue as possible as we can. CC."
   - This message lets you know Sec Antag is turned on. Heads should act against security from now 'officially.'
   - Meanwhile, even if this mode is turned on, none of security officers can be a traitor (because they didn't want to be a traitor or they just didn't get any chance to be a traitor at all.)
- They get none of Syndicate uplink because they need to be hideous, thus they get no TC. At least, they can get Syndicate Radio Encryption key upon spawn.
   - Unlike traitors from other departments, traitor security is really serious to handle and can make a serious impact more than other departments, so they are not allowed to have other syndicate items.

**Pull request closed without merging**